content/en/blog/_posts/2021-02-19-Kubernetes-Portability-Reimagined

```diff
@@ -0,0 +1,135 @@
+---
+layout: blog
+title: Kubernetes Portability Reimagined
+date:  2021-02-19
+slug:  Kubernetes-Portability-Reimagined
+url:   /blog/2021/02/Kubernetes-Portability-Reimagined
+---
+
+Kubernetes: Portability Reimagined
+
+Kubernetes provides a portable application architecture, and is delivered through a basic execution unit, Pod.  One or more containers are embedded into a Pod that carries key property for portability.  Kubernetes enables decoupling of environment specific configuration from container images along with abstractions for having applications run on on-premise or cloud infrastructures.
+
+Information related to application configuration such as key-value pairs are made available in the mode of configuration files or environment variables.  Such values may be non-confidential e.g., log-level, timeout specification and or sensitive data such as password, token.  Kubernetes objects, ConfigMap and Secret presents abstractions to declare and refer key-value pairs in a Pod thus help achieve application portability.
+
+Environment variables are likely to differ between deployments such as test and production.  These configuration parameters may be independently managed and scaled with the help of ConfigMap and Secrets resources, thus achieving separation of code from configuration data.  While Secrets provides basic security, base64 encoding; the ConfigMap and Secrets data is made available to nodes and Pods on need basis for the lifecycle of the Pod but not otherwise, although declared in the cluster.
+
+In the following example, we present a Pod, ConfigMap and Secrets that decouples the environment variables - counter, sleep-interval, username and password from embedding into a Container.  To refer these environment variables in a container of a Pod, one needs to perform the following steps
+
+Declare key-value pairs into a Kubernetes namespace using ConfigMap and Secrets object
+Refer to these key-value pairs as data volume files or environment variables in a Container of a Pod
+
+$ cat kube-configmap.yaml
+Declaration of 2
+Key-value pairs
```

Comment:
Hi. Can you put this inside a [code fence](https://www.markdownguide.org/extended-syntax/#fenced-code-blocks)? Also, do the same for other blocks of code in the page.